### PR TITLE
Issue #147: combine duplicate boxes at the generational level

### PR DESCRIPTION
**Pull request by fire-eggs**
1. For descendant chart.
2. Has no negative impact on existing charts.
3. For the specific case shown in issue #147, combines 'duplicate' boxes

Limitation: 'duplicate' boxes can only be combined at the same generational level. 'Duplicates' across generational rows are not combined with this commit.

Limitation: in the specific case shown in issue #147, the "combined" boxes are placed a little "off center". The resulting appearance is not as "nice" as it could be.

### CODE REVIEW

File: projects/GKCore/GKCore/Charts/TreeChartModel.cs [hunk 1]
```diff
@@ -1014,6 +1033,55 @@ private void RecalcDesc(TreeChartPerson person, ExtPoint aPt, bool predef)
...
+            }
+        }
+
+        private int FixOverlap(TreeChartPerson person)
+        {
+            // Kevin Routley (aka fire-eggs, aka KBR). Partially address issue #147.
+            // Check for, and correct, collisions of boxes in this person's 
+            // generational row.
+            int shiftAmount = 0;
+            for (int m = 0; m <= fPersons.Count; m++)
```
Comment (Serg-Norseman) [inline projects/GKCore/GKCore/Charts/TreeChartModel.cs]: This is not a mistake by accident?
`m <= fPersons.Count`

**Comment by Serg-Norseman** [comment]
I am sorry :( but... 
see the attached file, with descendants from "Sidorov Dmitriy Andreevich"
[FullFamilyTreeTests.zip](https://github.com/Serg-Norseman/GEDKeeper/files/1515448/FullFamilyTreeTests.zip)

**Comment by Serg-Norseman** [comment]
I'll try to think about how to rebuild the algorithm one of these days. Maybe something will turn out ...

File: projects/GKCore/GKCore/Charts/TreeChartModel.cs (hunk 1 above)
Comment (fire-eggs) [reply projects/GKCore/GKCore/Charts/TreeChartModel.cs]: Yup, it is. Good eye! [There is a loop short-circuit which appears to have prevented it from being an issue in during my testing.]

**Comment by fire-eggs** [comment]
Oh wow! This is definitely much harder than I figured ...

If I can make some time, I'll poke at this new test. In the meantime, it'd
be better to ignore my attempts to fix this. I need to establish a set of
test trees to exercise more variations on the problem.

Sorry for wasting your time ...
Kevin

**Comment by Serg-Norseman** [comment]
Kevin, it's okay! I am always glad to consider any additions to this project and I am happy with any participation :)

I still can not get and think about possible ways to implement the "Full Tree". Work takes up all resources. Perhaps in a week or two there will be additional free time. There are a couple of ideas on how to extend the mapping of branches under the current engine. However, these ideas do not yet relate to how to combine duplicate information.

**Pull request merged**